Tests/CredentialsJWTTests/TestRawRouteJWT.swift

```diff
@@ -0,0 +1,289 @@
+                XCTAssertEqual(response?.statusCode, HTTPStatusCode.OK, "HTTP Status code was \(String(describing: response?.statusCode))")
+                do {
+                    guard let body = try response?.readString() else {
+                        XCTFail("No response body")
+                        return
+                    }
+                    let profile = body
+                    let testProfile = UserProfile(id: "Test", displayName: "Test", provider: "JWT")
+                    XCTAssertEqual(profile, testProfile.id)
+                } catch {
+                    XCTFail("Could not decode response: \(error)")
+                }
+                expectation.fulfill()
+            }, headers: ["X-Token-Type" : "JWT", "Authorization" : "Bearer " + self.jwtString])
+        }
+    }
+
+    // Tests that when an incorrect token is supplied a valid token but wrong UserProfile is created.
+    func testIncorrectToken() {
+        performServerTest(router: router) { expectation in
+            self.performRequest(method: "get", path: "/rawtokenauth", callback: { response in
+                XCTAssertNotNil(response, "ERROR!!! ClientRequest response object was nil")
+                XCTAssertEqual(response?.statusCode, HTTPStatusCode.OK, "HTTP Status code was \(String(describing: response?.statusCode))")
+                do {
+                    guard let body = try response?.readString() else {
+                        XCTFail("No response body")
+                        return
+                    }
+                    let profile = body
+                    let testProfile = UserProfile(id: "Test", displayName: "Test", provider: "JWT")
+                    XCTAssertNotEqual(profile, testProfile.id)
+                } catch {
+                    XCTFail("Could not decode response: \(error)")
+                }
+                expectation.fulfill()
+            }, headers: ["X-Token-Type" : "JWT", "Authorization" : "Bearer " + self.jwtString2])
+        }
+    }
+
+    // Tests that when an incorrect token is supplied an invalid token is supplied, user is unauthorized.
+    func testInvalidToken() {
+            performServerTest(router: router) { expectation in
+                self.performRequest(method: "get", path: "/rawtokenauth", callback: { response in
+                    XCTAssertNotNil(response, "ERROR!!! ClientRequest response object was nil")
+                    XCTAssertEqual(response?.statusCode, HTTPStatusCode.unauthorized, "HTTP Status code was \(String(describing: response?.statusCode))")
+                    expectation.fulfill()
+                }, headers: ["X-Token-Type" : "JWT", "Authorization" : "Bearer " + "Wrong"])
+            }
+        }
+
+    // Tests that when a request to a raw route that includes this middleware does not
+    // contain the matching X-token-type header, the middleware skips authentication and the
+    // google handler is instead invoked.
+    func testGoogleTokenType() {
```

Comment:
I'd suggest renaming to `testSkipAuthentication` as it more accurately represents what the test is testing.